sig-scalability/processes/scalability-validation.md

```diff
@@ -0,0 +1,143 @@
+
+## Goals
+
+In this document, we address the following process-related problems wrt scale testing:
+
+- Automate large cluster tests with a reasonable frequency
+- Concretely define the testing configuration used for the release
+- Make scalability validation a release prerequisite
+- Clearly lay down responsibilities for scale tests
+
+## Non-Goals
+
+This document does not intend to:
+
+- Define the set of tests that comprise scalability and correctness suite
+- Define SLIs/SLOs (that’s discussed [here]) and thresholds for the tests
+- Discuss particular performance issues we’re facing wrt different releases
+
+While these are interesting from scalability perspective, they are not testing process issues but details of the tests themselves. And can change without requiring much changes in the process.
+
+## Proposed design
+
+For each of the problems above we propose solutions and discuss some caveats. This is an open discussion and better solutions may evolve in future.
+
+### Automate large cluster tests
+
+Currently, there are 2 kinds of tests needed to validate the claim “kubernetes supports X-node clusters”. These are:
+
+- Correctness tests - e2e tests verifying expected system behaviours
+- Performance tests - e2e tests for stress-testing the system perf ([Feature:Performance])
+
+We need to run them on 5k-node clusters, but they’re:
+
+- Time-consuming (12-24 hrs)
+- Expensive (tens of thousands of core hours per run)
+- Blocking other large tests (quota limitations + only one large test project available viz. 'kubernetes-scale')
+
+So we don’t want to run them too frequently. On the other hand, running them too infrequently means late identification and piling up of regressions. So we choose the following middleground:
+
+- Performance tests on 2k-node/5k-node GCE clusters alternatingly from Mon-Sat
+  - would give us one performance run from each day to help catch regressions fast
+  - running 2k-node on alternating days gives time for 5k-node correctness tests to run on those days
+  - many of the performance regressions on 5k-node should also be seen on 2k-node (albeit a smaller version probably)
+- Correctness tests on 2k-node/5k-node GCE clusters alternatingly from Mon-Sat
+  - would give us one correctness run from each day to help catch regressions fast
+  - running 2k-node on alternating days gives time for 5k-node performance tests to run on those days
+  - many of the correctness regressions on 5k-node should also be seen on 2k-node
+- Performance tests on 2k-node GKE cluster on Sun
+  - would give us a performance run for sunday too
+  - would also additionally help verify performance of GKE
+- Correctness tests on 2k-node GKE cluster on Sun
+  - would give us a correctness run for sunday too
+  - would also additionally help verify correctness of GKE
+
+Here's the proposed schedule (may be fine-tuned later based on test health / release schedule):
+(B = release-blocking job)
+
+| Day | | |
```

Comment:
We need to update the schedule, as this is no longer accurate. But it shouldn't block this PR, as this is just moving from one place to another.